driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -171,15 +183,33 @@ public void run() {
   // following threshold, we just replace the connection by a new one.
   private final int minAllowedStreams;
 
-  HostConnectionPool(Host host, HostDistance hostDistance, SessionManager manager) {
+  HostConnectionPool(Host host, HostDistance hostDistance, final SessionManager manager) {
     assert hostDistance != HostDistance.IGNORED;
     this.host = host;
     this.hostDistance = hostDistance;
     this.manager = manager;
 
     this.minAllowedStreams = options().getMaxRequestsPerConnection(hostDistance) * 3 / 4;
 
-    this.timeoutsExecutor = manager.getCluster().manager.connectionFactory.eventLoopGroup.next();
+    this.scheduler = manager.getCluster().manager.connectionFactory.eventLoopGroup.next();
+
+    scheduler.schedule(
+        new Runnable() {
+          @Override
+          public void run() {
+            if (!isClosed()) {
+              ConnectionTask task = pendingTasks.poll();
+              if (task != null) {
+                manager.blockingExecutor().submit(task);
+                scheduler.schedule(this, 1, TimeUnit.SECONDS);
```

Comment:
Instead of having an infinite loop like that we can schedule tasks directly on the scheduler with 100ms delay.